docs/deployment/deploy_gardenlet.md

```diff
@@ -0,0 +1,344 @@
+    
+2.  Create a bootstrap kubeconfig based on the bootstrap token created in the Garden cluster.
+    
+    ```yaml
+    apiVersion: v1
+    kind: Config
+    current-context: gardenlet-bootstrap@default
+    clusters:
+    - cluster:
+        certificate-authority-data: <ca-of-garden-cluster>
+        server: https://<endpoint-of-garden-cluster>
+      name: default
+    contexts:
+    - context:
+        cluster: default
+        user: gardenlet-bootstrap
+      name: gardenlet-bootstrap@default
+    users:
+    - name: gardenlet-bootstrap
+      user:
+        token: <bootstrap-token>
+    ```
+3.  Provide this bootstrap kubeconfig together with a desired name and namespace to the Gardenlet Helm chart values [here](../../charts/gardener/gardenlet/values.yaml#L31-L35):
+    
+    ```yaml
+    gardenClientConnection:
+      bootstrapKubeconfig:
+        name: gardenlet-kubeconfig-bootstrap
+        namespace: garden
+        kubeconfig: |
+          <bootstrap-kubeconfig>  # will be base64 encoded by helm
+    ```
+    
+    The bootstrap kubeconfig will be stored in the specified secret.
+    
+4.  Define a name and namespace where the Gardenlet shall store the real kubeconfig it creates during the bootstrap process [here](../../charts/gardener/gardenlet/values.yaml#L31-L35):
+    
+    This secret will be created by the Gardenlet in case it does not exist.
+    
+    ```yaml
+    gardenClientConnection:
+      kubeconfigSecret:
+        name: gardenlet-kubeconfig
+        namespace: garden
+    ```
+
+## Automatically register as a Seed cluster
+
+A Seed cluster can either be registered by manually creating the [`Seed` resource](https://github.com/gardener/gardener/blob/master/example/50-seed.yaml) or automatically by the Gardenlet.  
+This functionality is particularly useful for `Shooted Seed` cluster, as the Gardenlet in the Garden cluster deploys a copy of itself into the cluster with automatic registration of the `Seed` configured.  
+However it can also be used to have a streamlined Seed registration process when manually deploying the Gardenlet.
+
+**Prepare the Garden cluster**
+
+As a first step, the Garden cluster needs to be prepared.  
+Create a secret containing the Cloud Provider credentials of the Seed cluster.  
+This is required because the Gardenlet tries to create the `Seed` resource if it does not exist. This will fail if it references a missing secret.
+
+This is how the secret for AWS looks like:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: sweet-seed-aws
+  namespace: garden
+type: Opaque
+data:
+  accessKeyID: <base-64-access-key-id>
+  secretAccessKey: <base-64-access-key>
+```
+
+You can find examples for various providers [here](https://github.com/gardener/external-dns-management/tree/master/examples). The relevant example secrets are named `20-secret-<provider>-credentials`.
```

Comment:
Hm, why are you linking the external-dns-mgmt instead of the provider extensions?

Reply:
Linking the extension providers repo instead.